addons/userrights/userrightsprofile.class.inc.php

```diff
@@ -738,8 +738,12 @@ protected function GetUserActionGrant($oUser, $sClass, $iActionCode)
 		// load and cache permissions for the current user on the given class
 		//
 		$iUser = $oUser->GetKey();
-		$aTest = @$this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
-		if (is_array($aTest)) return $aTest;
+		if (isset($this->m_aObjectActionGrants[$iUser])
+			&& isset($this->m_aObjectActionGrants[$iUser][$sClass])
+			&& isset($this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode])){
+			$aTest = @$this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
+			if (is_array($aTest)) return $aTest;
```

Comment:
Only one `isset` is needed.
```suggestion
		if (isset($this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode])){
			$aTest = $this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
			if (is_array($aTest)) return $aTest;
```
Proof: https://3v4l.org/hYPZq